ui/app/AppLayouts/Profile/views/profile/MyProfileSettingsView.qml

```diff
@@ -69,20 +68,23 @@ ColumnLayout {
         }
 
         if (biometricsSwitch.checked && !biometricsSwitch.currentStoredValue)
-            Global.openPopup(storePasswordModal)
+            root.privacyStore.tryStoreToKeyChain()
         else if (!biometricsSwitch.checked)
-            localAccountSettings.storeToKeychainValue = Constants.keychain.storedValue.never;
+            root.privacyStore.tryRemoveFromKeyChain()
 
         reset()
     }
 
-    function offerToStorePassword(password, runStoreToKeyChainPopup)
-    {
-        if (Qt.platform.os !== "osx")
-            return;
+    Connections {
+        target: Qt.platform.os === Constants.mac? root.privacyStore.privacyModule : null
```

Comment:
Following on the above idea, might lead to sourcing `SUPPORTS_FINGERPRINT`  from backend here